### PR TITLE
DM-24829: Use a magic string for no component in datastore record

### PR DESCRIPTION
**Pull request by timj**
We were using an empty string but Oracle forces those
to be NULL and that breaks things.

### CODE REVIEW

**Review by MichelleGower — APPROVED** [review]
I did not search for database admin recommendations to see if they differ from this solution.  I started running ci_hsc_gen3 with Oracle using this ticket branch.   It has made it past running ISR at the time of this comment.  Dug around the output repo filenames a little bit to make sure this string didn't end up in a filename and couldn't find any.  Approving the changes for merging.

File: python/lsst/daf/butler/datastores/fileLikeDatastore.py [hunk 1]
```diff
@@ -62,6 +62,9 @@
 
 log = logging.getLogger(__name__)
 
+# String to use when a Python None is encountered
+NULLSTR = "__NULL_STRING__"
```
Comment (timj) [inline python/lsst/daf/butler/datastores/fileLikeDatastore.py:66]: @TallJimbo I am pondering whether this should be shorter. Will people want to use this constant elsewhere at some point?

Comment (TallJimbo) [reply python/lsst/daf/butler/datastores/fileLikeDatastore.py:66]: Let's not worry about it.  I think if we did need to use sentinal empty strings in more places, we'd move more than just the constant into `Database`, and I don't have any anticipated use cases of that right now.

**Review by TallJimbo — APPROVED** [review]
Looks good.

Thank goodness the f-string lint didn't affect multi-line strings where some of the lines have substitutions and others don't (where I vastly prefer to start all lines with `f`).

**Pull request merged**